search-app/src/cmr/search/data/metadata_retrieval/metadata_cache.clj

```diff
@@ -48,152 +45,124 @@
+  (let [incremental-since-refresh-date (str (clj-time.core/now))
+        concepts-tuples (cmn-coll-metadata-cache/fetch-collections-from-elastic context)
         new-cache-value (reduce #(merge %1 (concept-tuples->cache-map context %2))
                                 {}
                                 (partition-all 1000 concepts-tuples))
-        cache (c/context->cache context cache-key)]
-    (reset! (:cache-atom cache) new-cache-value)
-    (info "Metadata cache refresh complete. Cache Size:" (c/cache-size cache))))
-
-(defn all-cached-revision-format-maps
-  "Returns a sequence of all revision format maps in the cache sorted by concept id"
+        rcache (hash-cache/context->cache context cmn-coll-metadata-cache/cache-key)]
+    (hash-cache/set-value rcache
+                          cmn-coll-metadata-cache/cache-key
+                          cmn-coll-metadata-cache/incremental-since-refresh-date-key
+                          incremental-since-refresh-date)
+    (hash-cache/set-values rcache cmn-coll-metadata-cache/cache-key new-cache-value)
+    (info "Metadata cache refresh complete. Cache Size:"
+          (hash-cache/cache-size rcache cmn-coll-metadata-cache/cache-key))))
+
+(defn update-cache-job
+  "Updates the collection metadata cache by querying elastic search for updates since the
+  last time the cache was updated."
   [context]
-  (let [cache (deref (:cache-atom (c/context->cache context cache-key)))
-        maps (for [concept-id (sort (keys cache))]
-               (get cache concept-id))]
-    (seq maps)))
-
-(defconfig refresh-collection-metadata-cache-interval
-  "The number of seconds between refreshes of the collection metadata cache"
-  {:default (* 3600 8)
-   :type Long})
+    (info "Updating collection metadata cache")
+    (let [incremental-since-refresh-date (str (t/now))
+          concepts-tuples (cmn-coll-metadata-cache/fetch-updated-collections-from-elastic context)
+          new-cache-value (reduce #(merge %1 (concept-tuples->cache-map context %2))
+                                  {}
+                                  (partition-all 1000 concepts-tuples))
+          cache (hash-cache/context->cache context cmn-coll-metadata-cache/cache-key)]
+      (hash-cache/set-value cache
+                            cmn-coll-metadata-cache/cache-key
+                            cmn-coll-metadata-cache/incremental-since-refresh-date-key
+                            incremental-since-refresh-date)
+      (hash-cache/set-values cache
+                             cmn-coll-metadata-cache/cache-key
+                             new-cache-value)
+      (info "Metadata cache update complete. Cache Size:" (hash-cache/cache-size cache cmn-coll-metadata-cache/cache-key))))
 
 (defjob RefreshCollectionsMetadataCache
   [ctx system]
-  (refresh-cache {:system system}))
+  (when (string/includes? (str (type (get-in {:system system} [:system :embedded-systems :metadata-db :db])))
```

Comment:
I see a lot of type casting into string to do a comparison. I'm not sure its needed in all these cases as you can just check for the actual class.

Reply:
https://clojuredocs.org/clojure.core/instance_q

Reply:
Thank you!  I am replacing all of the places where I have used string comparison of the class with instance?

Reply:
Well, I wasn't able to use instance? as that created a circular dependency.  I created a function in cmr.common.cache to use the string comparison of the type, so at least now, the comparison is done in one place. Another option that I thought of was to put a key in each of the defrecords that implement the CmrCache protocol, but then when a new cache is added, someone will have to remember to add the same key into it.  This way no one has to remember to do anything.